yarl/__init__.py

```diff
@@ -157,12 +157,13 @@ def __new__(cls, val="", *, encoded=False, strict=None):
             else:
                 host = val.hostname
                 if host is None:
-                    raise ValueError("Invalid URL: host is required for abolute urls.")
+                    raise ValueError("Invalid URL: host is required for absolute urls")
 
                 try:
                     port = val.port
-                except ValueError:
-                    raise ValueError("Invalid URL: port can't be converted to integer")
+                except ValueError as e:
+                    msg = "Invalid URL: port can't be converted to integer"
```

Comment:
I don't think that an extra var is really needed here. You could just multiline the statement